pennylane/transforms/dynamic_one_shot.py

```diff
@@ -267,18 +277,28 @@ def measurement_with_no_shots(measurement):
             )
         if interface != "jax" and m.mv and not has_valid:
             meas = measurement_with_no_shots(m)
+        elif m.mv and active_qjit:
+            meas = gather_mcm_qjit(m, mcm_samples, is_valid)  # pragma: no cover
         elif m.mv:
             meas = gather_mcm(m, mcm_samples, is_valid)
         elif interface != "jax" and not has_valid:
             meas = measurement_with_no_shots(m)
             m_count += 1
         else:
-            result = [res[m_count] for res in results]
+            result = results[m_count]
             if not isinstance(m, CountsMP):
                 # We don't need to cast to arrays when using qml.counts. qml.math.array is not viable
                 # as it assumes all elements of the input are of builtin python types and not belonging
                 # to any particular interface
-                result = qml.math.stack(result, like=interface)
+                result = qml.math.array(result, like=interface)
+            if active_qjit:  # pragma: no cover
+                if isinstance(m, CountsMP):
+                    normalized_meas.append(
+                        (result[0][0], qml.math.sum(result[1] * is_valid.reshape((-1, 1)), axis=0))
+                    )
+                    m_count += 1
+                    continue
+                result = qml.math.squeeze(result)
```

Comment:
It would be helpful to explain such special handling with a comment :)

Reply:
Will do.